publish-mqtt/src/main.rs

```diff
@@ -107,37 +108,67 @@ async fn main() -> Result<(), Box<dyn Error + Send + Sync>> {
     Ok(())
 }
 
-fn node_id_name_for_sensor(
-    sensor: &BluetoothDevice,
-    sensor_names: &HashMap<String, String>,
-) -> Result<(String, String), Box<dyn Error>> {
-    let mac_address = sensor.get_address()?;
-    let node_id = mac_address.replace(":", "");
-    let node_name = sensor_names
-        .get(&mac_address)
-        .cloned()
-        .unwrap_or(mac_address);
-    Ok((node_id, node_name))
+#[derive(Debug)]
+struct Sensor {
+    device_path: String,
+    mac_address: String,
+    name: String,
+    last_update_timestamp: Instant,
+}
+
+impl Sensor {
+    pub fn new(
+        device: &BluetoothDevice,
+        sensor_names: &HashMap<String, String>,
+    ) -> Result<Self, Box<dyn Error>> {
+        let mac_address = device.get_address()?;
+        let name = sensor_names
+            .get(&mac_address)
+            .cloned()
+            .unwrap_or_else(|| mac_address.clone());
+        Ok(Self {
+            device_path: device.get_id(),
+            mac_address,
+            name,
+            last_update_timestamp: Instant::now(),
+        })
+    }
+
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    pub fn mac_address(&self) -> &str {
+        &self.mac_address
+    }
```

Comment:
(controversial) I don't like getters. Can we just make the fields public and trust people to not fuck with them?

Reply:
Done.